pennylane/transforms/defer_measurements.py

```diff
@@ -101,9 +113,309 @@ def null_postprocessing(results):
+    try:
+        # pylint: disable=import-outside-toplevel
+        import jax
+
+        from pennylane.capture import PlxprInterpreter
+        from pennylane.capture.primitives import cond_prim, ctrl_transform_prim, measure_prim
+    except ImportError:  # pragma: no cover
+        return None, None
+
+    # pylint: disable=redefined-outer-name
+
+    class DeferMeasurementsInterpreter(PlxprInterpreter):
+        """Interpreter for applying the defer_measurements transform to plxpr."""
+
+        # pylint: disable=unnecessary-lambda-assignment,attribute-defined-outside-init,no-self-use
+
+        def __init__(self, aux_wires):
+            super().__init__()
+            self._aux_wires = Wires(aux_wires)
+
+            # We use a dict here instead of a normal int variable because we want the state to mutate
+            # when we interpret higher-order primitives
+            self.state = {"cur_idx": 0}
+
+        def cleanup(self) -> None:
+            """Perform any final steps after iterating through all equations.
+
+            Blank by default, this method can clean up instance variables. Particularly,
+            this method can be used to deallocate qubits and registers when converting to
+            a Catalyst variant jaxpr.
+            """
+            self.state = {"cur_idx": 0}
+
+        def interpret_dynamic_operation(self, data, struct, idx):
+            """Interpret an operation that uses mid-circuit measurement outcomes as parameters.
+
+            * This will not work if mid-circuit measurement values are used to specify
+              operator wires.
+            * This will not work if more than one parameter uses mid-circuit measurement values.
+
+            Args:
+                data (TensorLike): Flattened data of the operator
+                struct (PyTreeDef): Pytree structure of the operator
+                idx (int): Index of mid-circuit measurement value in ``data``
+
+            Returns:
+                None
+            """
+            mv = data[idx]
+            for branch, value in mv.items():
+                data[idx] = value
+                op = jax.tree_util.tree_unflatten(struct, data)
+                qml.ctrl(op, mv.wires, control_values=branch)
+
+        def interpret_operation(self, op: "pennylane.operation.Operator"):
+            """Interpret a PennyLane operation instance.
+
+            Args:
+                op (Operator): a pennylane operator instance
+
+            Returns:
+                Any
+
+            This method is only called when the operator's output is a dropped variable,
+            so the output will not affect later equations in the circuit.
+
+            See also: :meth:`~.interpret_operation_eqn`.
+
+            """
+            data, struct = jax.tree_util.tree_flatten(op)
+
+            idx = -1
+            for i, d in enumerate(data):
+                if isinstance(d, MeasurementValue):
+                    idx = i
+                    break
```

Comment:
```suggestion
                    return self.interpret_dynamic_operation(data, struct, i)
```
?

Also, what about something like:
```
m0 = qml.measure(0)
m1 = qml.measure(1)
qml.Rot(m0, m1, 0.5, wires=0)
```
?

Reply:
I've currently documented in the `defer_measurements` program capture usage details that only ops with single parameters work with MCM inputs.

Reply:
The number of logical branches will blow up very fast. If you combine $n_i$ MCMs to create the i-th parameter for a gate, the number of controlled gates we will need to add for a gate with `m` MCM parameters is $2^{n_1} * 2^{n_2} * ... * 2^{n_m}$